src/Draco.Compiler/Internal/Documentation/DocumentationElement.cs

```diff
@@ -0,0 +1,115 @@
+using System.Collections.Immutable;
+using System.Linq;
+using System.Xml.Linq;
+using Draco.Compiler.Internal.Symbols;
+
+namespace Draco.Compiler.Internal.Documentation;
+
+/// <summary>
+/// Represents single documentation element.
+/// </summary>
+internal abstract record class DocumentationElement
+{
+    /// <summary>
+    /// Creates a markdown representation of this documentation element.
+    /// </summary>
+    /// <returns>The documentation in markdown format.</returns>
+    public abstract string ToMarkdown();
+
+    /// <summary>
+    /// Creates an XML representation of this documentation element.
+    /// </summary>
+    /// <returns>The documentation in XML format.</returns>
+    public abstract XNode ToXml();
```

Comment:
I'm confused why we support these on individual elements, if we have determined that we can't do it for XML for example

Reply:
As in, we could just have the conversion on the top-level documentation roots